frontend/packages/dev-console/src/components/edit-application/EditApplication.tsx

```diff
@@ -1,113 +1,105 @@
-import * as EditApplicationUtils from './edit-application-utils';
+import { getPageHeading, getInitialValues } from './edit-application-utils';
 
-const EditApplication: React.FC<EditApplicationProps> = ({
+export interface StateProps {
+  perspective: string;
+}
+
+const EditApplication: React.FC<EditApplicationProps & StateProps> = ({
+  perspective,
   namespace,
   appName,
-  editAppResource,
   resources: appResources,
-  onCancel,
-  onSubmit,
 }) => {
-  const builderImages: NormalizedBuilderImages =
+  const imageStreamsData =
     appResources.imageStreams && appResources.imageStreams.loaded
-      ? normalizeBuilderImages(appResources.imageStreams.data)
-      : null;
-
-  const currentImage = _.split(
-    _.get(appResources, 'buildConfig.data.spec.strategy.sourceStrategy.from.name', ''),
-    ':',
-  );
+      ? appResources.imageStreams.data
+      : [];
+  const builderImages: NormalizedBuilderImages = !_.isEmpty(imageStreamsData)
+    ? normalizeBuilderImages(imageStreamsData)
+    : null;
 
-  const appGroupName = _.get(editAppResource, 'metadata.labels["app.kubernetes.io/part-of"]');
+  const initialValues = getInitialValues(appResources, appName, namespace);
+  const pageHeading = getPageHeading(_.get(initialValues, 'build.strategy', ''));
 
-  const initialValues = {
-    formType: 'edit',
-    name: appName,
-    application: {
-      name: appGroupName,
-      selectedKey: appGroupName,
-    },
-    project: {
-      name: namespace,
-    },
-    git: EditApplicationUtils.getGitData(_.get(appResources, 'buildConfig.data')),
-    docker: {
-      dockerfilePath: _.get(
-        appResources,
-        'buildConfig.data.spec.strategy.dockerStrategy.dockerfilePath',
-        'Dockerfile',
-      ),
-      containerPort: parseInt(
-        _.split(_.get(appResources, 'route.data.spec.port.targetPort'), '-')[0],
-        10,
-      ),
-    },
-    image: {
-      selected: currentImage[0] || '',
-      recommended: '',
-      tag: currentImage[1] || '',
-      tagObj: {},
-      ports: [],
-      isRecommending: false,
-      couldNotRecommend: false,
-    },
-    route: EditApplicationUtils.getRouteData(_.get(appResources, 'route.data'), editAppResource),
-    resources: EditApplicationUtils.getResourcesType(editAppResource),
-    serverless: EditApplicationUtils.getServerlessData(editAppResource),
-    pipeline: {
-      enabled: false,
-    },
-    build: EditApplicationUtils.getBuildData(_.get(appResources, 'buildConfig.data')),
-    deployment: EditApplicationUtils.getDeploymentData(editAppResource),
-    labels: EditApplicationUtils.getUserLabels(editAppResource),
-    limits: EditApplicationUtils.getLimitsData(editAppResource),
+  const handleRedirect = (project: string) => {
+    const perspectiveData = plugins.registry
+      .getPerspectives()
+      .find((item) => item.properties.id === perspective);
+    const redirectURL = perspectiveData.properties.getImportRedirectURL(project);
+    history.push(redirectURL);
```

Comment:
This function can be moved into a util file(maybe `import-submit-utils` and can be used in both in `ImportForm` and here as well, instead of having same code in both places.